grakn-core/src/main/java/ai/grakn/util/CommonUtil.java

```diff
@@ -160,4 +162,47 @@ public static RuntimeException unreachableStatement(@Nullable String message, Th
             }
         };
     }
+
+    public static <T, K, V> Collector<T, ?, HashMultimap<K, V>> flatteningToMultimap(
+            Function<? super T, ? extends K> keyFunction,
+            Function<? super T, ? extends Stream<? extends V>> valueFunction) {
+        return flatteningToMultimap(keyFunction, valueFunction, HashMultimap::create);
+    }
+
+    public static <T, K, V, M extends Multimap<K, V>> Collector<T, ?, M> flatteningToMultimap(
```

Comment:
I've re-discovered the incredible `for` loop, so I'll get rid of this scary method.